flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/schema/DeferredTypeFlinkTableFunction.scala

```diff
@@ -40,6 +43,10 @@ class DeferredTypeFlinkTableFunction(
     val implicitResultType: TypeInformation[_])
   extends FlinkTableFunction(tableFunction, evalMethod) {
 
+  val paramTypeInfos = evalMethod.getParameterTypes.map { paramType =>
+    TypeExtractor.getForClass(paramType)
```

Comment:
I think signature types of the `eval()` method will be better. First, It will make them no differences in both situations.  I think I'd like to try to do convert the first case to the signatures of the `eval()` method. It will make it easier to identify which method is called.